From beff9090fd66952cbf30337d8813f3e552749d5d Mon Sep 17 00:00:00 2001
From: Jon Medhurst <tixy@linaro.org>
Date: Tue, 17 Sep 2013 15:29:33 +0100
Subject: [PATCH] drivers: CCI: Disable probing when PSCI is being used.

When booting with boottime selection of PSCI the device-tree still
contains a CCI device, even though this isn't accessible in the
non-secure world. We therefore need to detect this situation, otherwise
cci_pmu_probe will call probe_cci_revision and crash the system on boot.

Signed-off-by: Jon Medhurst <tixy@linaro.org>
(cherry picked from commit 9c6756fbc53e880dfcbd091737d9f116435b99a4)
Signed-off-by: Alex Shi <alex.shi@linaro.org>
---
 drivers/bus/arm-cci.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/bus/arm-cci.c b/drivers/bus/arm-cci.c
index 962fd35..8965fb5 100644
--- a/drivers/bus/arm-cci.c
+++ b/drivers/bus/arm-cci.c
@@ -26,6 +26,7 @@
 
 #include <asm/cacheflush.h>
 #include <asm/irq_regs.h>
+#include <asm/psci.h>
 #include <asm/pmu.h>
 #include <asm/smp_plat.h>
 
@@ -969,6 +970,11 @@ static int cci_probe(void)
 	const char *match_str;
 	bool is_ace;
 
+	if (psci_probe() == 0) {
+		pr_debug("psci found. Aborting cci probe\n");
+		return -ENODEV;
+	}
+
 	np = of_find_matching_node(NULL, arm_cci_matches);
 	if (!np)
 		return -ENODEV;
-- 
1.8.0

